src/test/java/hudson/maven/MavenJenkinsRule.java

```diff
@@ -23,4 +29,16 @@ public void before() throws Throwable {
         jenkins.getDescriptorByType(MavenModuleSet.DescriptorImpl.class).setGlobalMavenOpts(JAVA_HEADLESS_OPT);
     }
 
+    /**
+     * Fix the focus issue when a JVM is created to launch a slave by adding {@link #JAVA_HEADLESS_OPT}
+     */
+    @Override
+    public CommandLauncher createComputerLauncher(EnvVars env) throws URISyntaxException, MalformedURLException {
+        int sz = this.jenkins.getNodes().size();
+        return new CommandLauncher(String.format("\"%s/bin/java\" %s %s -jar \"%s\"",
+                new Object[]{System.getProperty("java.home"), JAVA_HEADLESS_OPT, SLAVE_DEBUG_PORT > 0 ?
```

Comment:
@jglick @abayer @daniel-beck maybe this change could be useful for the original JenkinsRule ? I imagine that for any plugin creating a slave you have the same issue for developers on MacOS X